jvm-libs/test-utils/src/main/kotlin/maru/test/genesis/BesuGenesisFactory.kt

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * This file is dual-licensed under either the MIT license or Apache License 2.0.
+ * See the LICENSE-MIT and LICENSE-APACHE files in the repository root for details.
+ *
+ * SPDX-License-Identifier: MIT OR Apache-2.0
+ */
+package maru.test.genesis
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.ObjectNode
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import maru.consensus.DifficultyAwareQbftConfig
+import maru.consensus.ElFork
+import maru.consensus.ForksSchedule
+import org.hyperledger.besu.tests.acceptance.dsl.node.configuration.genesis.GenesisConfigurationFactory
+
+class BesuGenesisFactory(
+  val genesisTemplate: String = genesisTemplateLondonWithoutConsensus,
+  val cliqueBlockTimeSeconds: UInt = 1u,
+  val cliqueEmptyBlocks: Boolean = true,
+) {
+  private var forkSchedule: ForksSchedule? = null
+
+  // we can cache the genesis if for the same fork schedule
+  // which does not change per cluster (for now at least)
+  // building the genesis for many nodes repeatedly is wasteful
+  private var cachedGenesis: String? = null
+
+  fun setForkSchedule(value: ForksSchedule) {
+    this.forkSchedule = value
+    // invalidate cached genesis when fork schedule changes
+    cachedGenesis = null
+  }
+
+  fun create(): String {
+    if (this.forkSchedule == null) {
+      throw IllegalStateException("forkSchedule must be initialized before calling creating genesis")
+    }
+
+    return if (cachedGenesis != null) {
+      return cachedGenesis!!
+    } else {
+      cachedGenesis =
+        createGenesisWithClique(
+          genesisTemplate,
+          cliqueBlockTimeSeconds,
+          cliqueEmptyBlocks,
+          forkSchedule!!,
+        )
+      println("new genesis created: $cachedGenesis")
+      cachedGenesis!!
+    }
+  }
+
+  companion object {
+    val genesisTemplateLondonWithoutConsensus: String =
+      GenesisConfigurationFactory.readGenesisFile("/besu-genesis-template.json")
+    private val jsonObjectMapper: ObjectMapper = jacksonObjectMapper()
+
+    fun createGenesisWithClique(
```

Comment:
I suggest to redesign this slightly:
1. make this method private
2. Make this method to return a JsonNode instead of a String

Then we can treat `JsonNode` as a Builder object and make it look even more like a builder if we change turn `setGenesisConfigProperty` into an extension method and make it return a JsonNode object